scheduler/src/cook/mesos/constraints.clj

```diff
@@ -114,6 +114,18 @@
   (let [needs-gpus? (job-needs-gpus? job)]
     (->gpu-host-constraint job needs-gpus?)))
 
+(defrecord rebalancer-reservation-constraint [reserved-hosts]
+  JobConstraint
+  (job-constraint-name [this] (get-class-name this))
+  (job-constraint-evaluate
+    [this _ vm-attributes]
+    (let [target-hostname (get vm-attributes "HOSTNAME")]
+      [(not (contains? reserved-hosts target-hostname))
+       "Host is reserved for rebalancing"]))
```

Comment:
Where does this string get consumed/exposed?

Reply:
This will be recorded and shown on the `unscheduled_jobs` endpoint if triggered by a user.

Reply:
Are there cases where I would call `unscheduled_jobs` and see a bunch of "Host is reserved for rebalancing" messages? If so, that seems like it would be confusing

Reply:
These would be aggregated by host, so you'd see something like:
`[{:reason \"Not enough CPU available\" :host_count 2}
   {:reason \"Not enough Memory available\" :host_count 3}
   {:reason \"Job already ran on this host.\" :host_count 4}
   {:reason \"Host is reserved for rebalancing\" :host_count 4}]`

I'm happy to change the message if you can think of something better

Reply:
Ah, got it, that makes sense.